validphys2/src/validphys/pseudodata.py

```diff
@@ -235,6 +237,154 @@ def indexed_make_replica(groups_index, make_replica):
     return pd.DataFrame(make_replica, index=groups_index, columns=["data"])
 
 
+def level0_commondata_wc(data, fakepdf):
+    """
+    Given a validphys.core.DataGroupSpec object, load commondata and
+    generate a new commondata instance with central values replaced
+    by fakepdf prediction
+
+    Parameters
+    ----------
+
+    data : validphys.core.DataGroupSpec
+
+    fakepdf: str
+
+    Returns
+    -------
+    list
+        list of validphys.coredata.CommonData instances corresponding to
+        all datasets within one experiment. The central value is replaced
+        by Level 0 fake data.
+
+    Example
+    -------
+    >>> from validphys.api import API
+    >>> API.level0_commondata_wc(dataset_inputs=[{"dataset":"NMC"}], use_cuts="internal", theoryid=200, fakepdf="NNPDF40_nnlo_as_01180")
+
+    [CommonData(setname='NMC', ndata=204, commondataproc='DIS_NCE', nkin=3, nsys=16)]
+    """
+
+    level0_commondata_instances_wc = []
+
+    # argument fakepdf is passed as str
+    # generate a core.PDF object with name equal to fakepdf argument
+    fakepdf = PDF(name=fakepdf)
```

Comment:
This could be done by changing the input instead.
I don't think that it makes sense in general for `fakepdf` to be a string, it might be better for it to be a `core.PDF` object.
To do so you can have a look at the parse rules in `validphys.config.py` module.
In particular, if I am not mistaken, it should be already implemented that way within nnpdf.

Note, however, that this has to be tested, as there might be some parts of the SIMUnet code that still assume that `fakepdf` is a string

Reply:
I added `parse_fakepdf` to `config.py`. It does what we want, _i.e._ it parse the fakepdf in the runcard as a `Core.PDF` object, I have to test in other scenarios, _e.g._ closuretests